.github/workflows/quickstart.yaml

```diff
@@ -0,0 +1,77 @@
+name: Build and Deploy
+
+on: [push, pull_request]
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
+
+      - name: Install dependencies
+        run: |
+          # Install kind
+          curl -sLo kind "$(curl -sL https://api.github.com/repos/kubernetes-sigs/kind/releases/latest | jq -r '[.assets[] | select(.name == "kind-linux-amd64")] | first | .browser_download_url')"
+          chmod +x kind
+          sudo mv kind /bin/
+          
+          # Install kubectl
+          curl -LO "https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl"
+          chmod +x kubectl
+          sudo mv kubectl /bin/
+          
+      - name: Build Docker image
+        run: |
+          sudo -E docker build -t checkpoint-restore-operator:ci .
+
+      - name: Delete existing Kubernetes cluster (if any)
+        run: |
+          kind delete cluster || true
+
+      - name: Create Kubernetes cluster
+        run: |
+          kind create cluster
+          kind export kubeconfig
+
+      - name: Load Docker image into kind cluster
+        run: |
+          kind load docker-image checkpoint-restore-operator:ci
+
+      - name: Deploy to Kubernetes
+        run: |
+          make install
+          make deploy IMG=checkpoint-restore-operator:ci
+
+      - name: Wait for deployments to be ready
+        run: |
+          wait_for_deployment() {
+            local deployment_name=$1
+            timeout=60
+            i=1
+            echo "Checking if the ${deployment_name} deployment is ready"
+            until kubectl -n checkpoint-restore-operator-system get deployment ${deployment_name} -o jsonpath='{.status.conditions[?(@.status=="True")].type}' | grep "Available" 2>/dev/null; do
+                ((i++))
+                if [[ ${i} -gt ${timeout} ]]; then
+                    echo "The ${deployment_name} deployment has not become ready before the timeout period"
+                    echo "Fetching deployment status and describing pods for debugging:"
+                    kubectl -n checkpoint-restore-operator-system get deployment ${deployment_name}
+                    kubectl -n checkpoint-restore-operator-system describe deployment ${deployment_name}
+                    kubectl -n checkpoint-restore-operator-system get pods
+                    kubectl -n checkpoint-restore-operator-system describe pods
+                    exit 1
+                fi
+                echo "Waiting for ${deployment_name} deployment to report a ready status"
+                sleep 5
+            done
+            echo "The ${deployment_name} deployment is ready"
+          }
+          wait_for_deployment checkpoint-restore-operator-controller-manager
```

Comment:
It would be better to have a script under `test` with these contents and invoke the script from the CI. Going forward, if we decide to make this any more complicated, the YAML will become unreadable.

Reply:
Got it, but what about this - 

> May be assuring only the ready state of checkpoint-restore-operator-controller-manager deployment is not 
> enough. Let me know any other things we need to assure for the operator deployment.

any thoughts?

Reply:
> any thoughts?

This is a good starting point. We can add more comprehensive tests later in another pull request.